real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/logging/OldLogConverter.java

```diff
@@ -0,0 +1,169 @@
+		String[] dateymd = {stdate.nextToken(), stdate.nextToken(), stdate.nextToken()};
+		String date_recorded = dateymd[1] + "/" + dateymd[2] + "/" + dateymd[0]; //This should now be in mm/dd/yyyy format
+		String timestamp = st2.nextToken();
+		
+		//Add heading
+		String logname = "Robobuggy Data Logs";
+		String schema_version = "1.0";
+		String software_version = "1.0.0";
+		writer.println("{");
+		writer.println("    \"name\": \"" + logname + "\",");
+		writer.println("    \"schema_version\": " + schema_version + ",");
+		writer.println("    \"date_recorded\": \"" + date_recorded + "\",");
+		writer.println("    \"software_version\": \"" + software_version + "\",");
+		writer.println("    \"sensor_data\": [");
+		
+		//Write the first set of data we read in before
+		writeSensorData(scanner, writer, st, type, timestamp);
+		
+		//Main loop to log sensor data
+		while(scanner.hasNextLine()){
+			s = scanner.nextLine();
+			st = new StringTokenizer(s, ",");
+			type = st.nextToken().substring(8); //Could be IMU, GPS, encoder, steering
+			st2 = new StringTokenizer(st.nextToken(), " ");
+			timestamp = st2.nextToken() + " " + st2.nextToken(); //Date time
+			writeSensorData(scanner, writer, st, type, timestamp);
+		}
+		//End sensor data part
+		writer.println("    ],");
+		
+		//Print the breakdown
+		writer.print("    \"data_breakdown\" : {");
+		for(int x = 0; x < sensor_type.size(); x++){
+			writer.print("\"" + sensor_type.get(x) + "\":" + sensor_quant.get(x));
+			//Omit last comma
+			if(x < sensor_type.size()-1){
+				writer.print(",");
+			}
+		}
+		writer.println("}");
+		
+		//Clean up and end
+		writer.print("}");
+		System.out.println("Done converting");
+	}
+	
+	//Update the types of sensors we have readings for as well as the number of readings
+	public static void updateDataBreakdown(ArrayList<String> sensor_type, ArrayList<Integer> sensor_quant, String type){
+		//If you don't have the sensor, add it to the list and note that you now have 1 reading for it
+		if(sensor_type.indexOf(type) == -1){
+			sensor_type.add(type);
+			sensor_quant.add(1);
+		}
+		//Otherwise, increment the number of readings you have
+		else{
+			sensor_quant.set(sensor_type.indexOf(type), sensor_quant.get(sensor_type.indexOf(type)).intValue()+1);
+		}
+	}
+	
+	//Given the necessary data, write it to a file in the new format. Lots of brute-force and case statements here
+	//Note: Anything that would be a tab is a quadruple space
+	//Another note: Capitalization of sensor names is off
+	public static void writeSensorData(Scanner scanner, PrintStream writer, StringTokenizer st, String name, String timestamp){
+		if(name.equalsIgnoreCase("GPS")){name = "GPS";}
+		if(name.equalsIgnoreCase("IMU")){name = "IMU";}
+		if(name.equals("logging_button")){name = "logging button";}
+		else{name = name.substring(0, 1).toUpperCase() + name.substring(1);}
+		writer.print("        {\"name\":\"" + name + "\",");
+		switch(name){
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [A switch statement does not contain a break](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1331974238/issues/source?bid=2938665&fileBranchId=2957575#l110)